app/src/main/java/com/randomappsinc/studentpicker/choosing/NameChoosingHistoryManager.java

```diff
@@ -0,0 +1,55 @@
+package com.randomappsinc.studentpicker.choosing;
+
+import android.content.Context;
+
+import androidx.annotation.NonNull;
+
+import com.afollestad.materialdialogs.MaterialDialog;
+import com.randomappsinc.studentpicker.R;
+import com.randomappsinc.studentpicker.models.ListInfo;
+import com.randomappsinc.studentpicker.utils.NameUtils;
+import com.randomappsinc.studentpicker.utils.UIUtils;
+
+public class NameChoosingHistoryManager {
+
+    public interface Listener {
+        void onEmptyHistoryNames();
+    }
+
+    private ListInfo listInfo;
+    private String namesHistory;
```

Comment:
This member variable isn't necessary, since the names history is always changing and we always just grab it on demand and immediately render it (local usage).